src/main/java/io/lakefs/routerfs/RouterFileSystem.java

```diff
@@ -120,7 +127,13 @@ public FSDataOutputStream append(Path f, int bufferSize, Progressable progress)
      */
     @Override
     public boolean rename(Path src, Path dst) throws IOException {
-        return false;
+        FileSystemPathTuple srcTuple = generateFSPathTuple(src);
+        FileSystemPathTuple dstTuple = generateFSPathTuple(dst);
+        if(!srcTuple.getFileSystem().equals(dstTuple.getFileSystem())) {
```

Comment:
I could not find a useful "equals" method for `FileSystem`s.  This may well cause issues!

Reply:
You're right.
Will compare the FS' URIs